proto/aserto/directory/reader/v3/reader.proto

```diff
@@ -0,0 +1,359 @@
+
+    // object methods
+    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {
+        option (google.api.http) = {
+            get: "/api/v3/directory/object/{object_type}/{object_id}"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get object instance"
+            description: "Returns single object instance, optionally with relations."
+            operation_id: "directory.v3.object.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };
+    };
+
+    rpc GetObjectMany(GetObjectManyRequest) returns (GetObjectManyResponse) {};
+    rpc GetObjects(GetObjectsRequest) returns (GetObjectsResponse) {
+        option (google.api.http) = {
+            get: "/api/v3/directory/objects/{object_type}"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "List object instances"
+            description: "Returns list of object instances."
+            operation_id: "directory.v3.objects.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };
+    };
+
+    // relation methods
+    rpc GetRelation(GetRelationRequest) returns (GetRelationResponse) {
+        option (google.api.http) = {
+            get: "/api/v3/directory/relation/{object_type}/{object_id}/{relation}/{subject_type}/{subject_id}/{subject_relation}"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get relation instance"
+            description: "Returns single relation instance, optionally with objects."
+            operation_id: "directory.v3.relation.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };
+    };
+
+    rpc GetRelations(GetRelationsRequest) returns (GetRelationsResponse) {
+        option (google.api.http) = {
+            get: "/api/v3/directory/relations/object/{object_type}/{object_id}/{relation}/{subject_type}/{subject_id}/{subject_relation}"
```

Comment:
Here too, the OpenAPI spec shows all path parameters as required but I don't think that's the intention.

Reply:
Same answer as above

Reply:
Does that mean that the gRPC REST gateway will correctly convert a request to `.../relations/object/user/ronen` to a `GetRelationsRequest` with empty strings for all parameters except `object_type` and `object_id`?

Reply:
Based on the conversation with Omri we remote GET relations in favor of a POST